### ISSUE TITLE
`consistent-conditional-object-spread`: Report unnecessary boolean casts in `logical` mode

### ISSUE THREAD

**Issue by abrahamguo** [sindresorhus/eslint-plugin-unicorn#3757]
### Description

I think that the rule should check and report on `{ ...!!cond && { a: 1 } }` as well, when the rule is in `logical` mode.

I proposed this to ESLint core's [no-extra-boolean-cast rule](https://eslint.org/docs/latest/rules/no-extra-boolean-cast) in [21328](https://github.com/eslint/eslint/issues/21328), but it was rejected.

### Examples

```js
// ❌
const object = {...!!condition && {property}};

// ✅
const object = {...condition && {property}};
```
